lerobot/scripts/eval.py

```diff
@@ -116,6 +116,7 @@ def rollout(
     Returns:
         The dictionary described above.
     """
+    assert isinstance(policy, nn.Module), "Policy must be a PyTorch nn module."
```

Comment:
Due to the way `Policy` was defined, we can't guarantee that anyone who implements the `Policy` protocol will also implement/inherit `nn.Module`. My goal for this PR is to make sure if we fail due wrong types we fail early, but there is a better way to solving this problem which I thought it needs to be done in a separate PR (`Policy` class inheriting `nn.module`)

Reply:
Hmm, okay, I mean I'm not 100% convinced because I'd say there are other things on a similar level that we are lax about checking. But I'm okay with this. To be clear though, if someone contributed some other function/class that uses a policy and _didn't_ put this check in, I wouldn't raise it in a review.